### PR TITLE
feature: support enable/disable action for resource alicloud_ess_alarm.

### PR DESCRIPTION
**Pull request by TalentNing**
TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudEssAlarm -timeout=300m 
=== RUN   TestAccAlicloudEssAlarm_basic
--- PASS: TestAccAlicloudEssAlarm_basic (141.60s)
=== RUN   TestAccAlicloudEssAlarm_multi
--- PASS: TestAccAlicloudEssAlarm_multi (59.46s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     201.128s

### CODE REVIEW

File: alicloud/resource_alicloud_ess_alarm.go [hunk 1]
```diff
@@ -250,6 +279,29 @@ func resourceAliyunEssAlarmUpdate(d *schema.ResourceData, meta interface{}) erro
...
+	d.SetPartial("enable")
+	if enable.(bool) {
+		enableAlarmRequest := ess.CreateEnableAlarmRequest()
+		enableAlarmRequest.AlarmTaskId = d.Id()
+		raw, err = client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+			return essClient.EnableAlarm(enableAlarmRequest)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), enableAlarmRequest.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
```
Comment (YY-tomorrow) [inline alicloud/resource_alicloud_ess_alarm.go]: please add addDebug()

File: alicloud/resource_alicloud_ess_alarm.go [hunk 2]
```diff
@@ -250,6 +279,29 @@ func resourceAliyunEssAlarmUpdate(d *schema.ResourceData, meta interface{}) erro
...
+	} else {
+		disableAlarmRequest := ess.CreateDisableAlarmRequest()
+		disableAlarmRequest.AlarmTaskId = d.Id()
+		raw, err = client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+			return essClient.DisableAlarm(disableAlarmRequest)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), disableAlarmRequest.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
```
Comment (YY-tomorrow) [inline alicloud/resource_alicloud_ess_alarm.go]: still

File: website/docs/r/ess_alarm.html.markdown [hunk 3]
```diff
@@ -84,6 +84,7 @@ The following arguments are supported:
 
 * `name` - (Optional) The name for ess alarm.
 * `description` - (Optional) The description for the alarm.
+* `enable` - (Optional) If enable specific ess alarm. Default to true.
```
Comment (YY-tomorrow) [inline website/docs/r/ess_alarm.html.markdown]: add available version

File: alicloud/resource_alicloud_ess_alarm.go [hunk 4]
```diff
@@ -250,6 +279,29 @@ func resourceAliyunEssAlarmUpdate(d *schema.ResourceData, meta interface{}) erro
 		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
 	}
 	addDebug(request.GetActionName(), raw)
+	enable := d.Get("enable")
```
Comment (YY-tomorrow) [inline alicloud/resource_alicloud_ess_alarm.go]: check haschange?

File: alicloud/resource_alicloud_ess_alarm.go (hunk 1 above)
Comment (TalentNing) [reply alicloud/resource_alicloud_ess_alarm.go]: fixed

File: website/docs/r/ess_alarm.html.markdown (hunk 3 above)
Comment (TalentNing) [reply website/docs/r/ess_alarm.html.markdown]: fixed

File: alicloud/resource_alicloud_ess_alarm.go (hunk 4 above)
Comment (TalentNing) [reply alicloud/resource_alicloud_ess_alarm.go]: fixed

File: alicloud/resource_alicloud_ess_alarm.go (hunk 2 above)
Comment (TalentNing) [reply alicloud/resource_alicloud_ess_alarm.go]: fixed

File: alicloud/resource_alicloud_ess_alarm.go [hunk 5]
```diff
@@ -142,6 +147,19 @@ func resourceAliyunEssAlarmCreate(d *schema.ResourceData, meta interface{}) erro
...
+	// enable or disable alarm
+	enable := d.Get("enable")
+	if !enable.(bool) {
+		disableAlarmRequest := ess.CreateDisableAlarmRequest()
+		disableAlarmRequest.AlarmTaskId = response.AlarmTaskId
+		raw, err = client.WithEssClient(func(essClient *ess.Client) (interface{}, error) {
+			return essClient.DisableAlarm(disableAlarmRequest)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, "alicloud_ess_alarm", disableAlarmRequest.GetActionName(), AlibabaCloudSdkGoERROR)
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_ess_alarm.go]: I think the line 159 should be `			return WrapErrorf(err, DefaultErrorMsg, d.Id(), disableAlarmRequest.GetActionName(), AlibabaCloudSdkGoERROR)
`

File: alicloud/resource_alicloud_ess_alarm.go [hunk 6]
```diff
@@ -242,6 +271,7 @@ func resourceAliyunEssAlarmUpdate(d *schema.ResourceData, meta interface{}) erro
 		}
 	}
 	request.Dimension = &createAlarmDimensions
+	d.SetPartial("dimensions")
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_ess_alarm.go]: `d.SetPartial` should be put after the API returned successfully.

File: alicloud/resource_alicloud_ess_alarm.go [hunk 7]
```diff
@@ -250,6 +280,32 @@ func resourceAliyunEssAlarmUpdate(d *schema.ResourceData, meta interface{}) erro
 		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
 	}
 	addDebug(request.GetActionName(), raw)
+	if d.HasChange("enable") {
+		enable := d.Get("enable")
+		d.SetPartial("enable")
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_ess_alarm.go]: The line should be moved in the line 307

File: website/docs/r/ess_alarm.html.markdown [hunk 8]
```diff
@@ -84,6 +84,7 @@ The following arguments are supported:
 
 * `name` - (Optional) The name for ess alarm.
 * `description` - (Optional) The description for the alarm.
+* `enable` - (Optional, Available in 1.48.0+) If enable specific ess alarm. Default to true.
```
Comment (xiaozhu36) [inline website/docs/r/ess_alarm.html.markdown]: `If enable specific ess alarm. Default to true.` should be `Whether to enable the specified ESS alarm. Default to true.`

File: alicloud/resource_alicloud_ess_alarm.go (hunk 5 above)
Comment (TalentNing) [reply alicloud/resource_alicloud_ess_alarm.go]: fixed

File: alicloud/resource_alicloud_ess_alarm.go (hunk 6 above)
Comment (TalentNing) [reply alicloud/resource_alicloud_ess_alarm.go]: fixed

File: website/docs/r/ess_alarm.html.markdown (hunk 8 above)
Comment (TalentNing) [reply website/docs/r/ess_alarm.html.markdown]: fixed

**Pull request merged**